libpod/container_log_linux.go

```diff
@@ -152,17 +152,22 @@ func (c *Container) readFromJournal(ctx context.Context, options *logs.LogOption
 
 			if !afterTimeStamp {
 				entryTime := time.Unix(0, int64(entry.RealtimeTimestamp)*int64(time.Microsecond))
-				if entryTime.Before(options.Since) {
+				if entryTime.Before(options.Since) && options.Until.IsZero() {
 					continue
 				}
 				afterTimeStamp = true
+				beforeTimeStamp = true
 			}
-			if beforeTimeStamp {
+
+			if !beforeTimeStamp {
```

Comment:
Don't we need to check "before" for each `entry` and exit once we passed it?

Reply:
My thinking here is that if we are before the time stamp we want to print the log line. From what I can tell, the current implementation of `since` causes you to skip anything that doesn't apply to you. I was trying to mimic that here and if we are before the `until` flag then that line does in fact apply to us.